docs/roleassumption.md

```diff
@@ -0,0 +1,420 @@
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+          volumeMounts:
+            - mountPath: /etc/ssl/certs
+              name: ssl-certs
+            - mountPath: /etc/kiam/tls
+              name: tls
+            - mountPath: /var/run/xtables.lock
+              name: xtables
+          livenessProbe:
+            httpGet:
+              path: /ping
+              port: 8181
+            initialDelaySeconds: 3
+            periodSeconds: 3
+```
+
+server-rbac.yaml
+
+```
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: kiam-server
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: kiam-read
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  - pods
+  verbs:
+  - watch
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: kiam-read
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kiam-read
+subjects:
+- kind: ServiceAccount
+  name: kiam-server
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: kiam-write
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: kiam-write
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kiam-write
+subjects:
+- kind: ServiceAccount
+  name: kiam-server
+  namespace: kube-system
+```
+
+After deploying the above components make sure that the kiam_server and kiam_agent pods are up and running. 
+
+## 4. Create the target cluster
+Make sure you create copy of the "aws/out" directory called "out2". To create the target cluster we must update the provider_components.yaml generated in the out2 directory as shown below (to be run from the repository root directory):
+
+```
+cp cmd/clusterctl/examples/aws/out cmd/clusterctl/examples/aws/out2
+vi cmd/clusterctl/examples/aws/out2/provider-components.yaml
+```
+
+1. Remove the credentials secret added at the bottom of the provider_components.yaml and do not mount the secret 
+2. Add the following annotation to the template of aws-provider-controller-manager stateful set to specify the new role that was created in target account.
+```
+      annotations:
+        iam.amazonaws.com/role: arn:aws:iam::<TARGET_AWS_ACCOUNT>:role/cluster-api
+```
+3. Also add this below annotation to the "aws-provider-system" namespace 
+```
+  annotations:
+    iam.amazonaws.com/permitted: ".*"
+```
+
+Create a new cluster using the steps similar to the one used to create the source cluster. They are as follows:
+```
+export SOURCE_KUBECONFIG=<PATH_TO_SOURCE_CLUSTER_KUBECONFIG>
+
+kubectl alpha phases apply-cluster-api-components --provider-components cmd/clusterctl/examples/aws/out2/provider-components.yaml \
```

Comment:
Instead of deploying the provider components a second time, it might be better to just edit the deployed provider components:

```sh
kubectl -n aws-provider-system edit statefulset aws-provider-controller-manager
# manually remove the referenced secret and save
kubectl -n aws-provider-system delete secret <secret name>
```

This will make it more clear the changes being made to the provider components, and it will also avoid leaving the bootstrapping secret in the deployed cluster, since updating the provider components will not delete the secret.

Reply:
I am not sure if I follow you correctly. The provider components is deployed for the second time on a different cluster running in AWS. The first time its deployed on the local cluster created by kind.

Reply:
doh, that is correct. My apologies.